### PR TITLE
[WIP] GRAM: pin path

### PR DESCRIPTION
**Pull request by vlad20012**
Better parser recovery for incomplete paths

### CODE REVIEW

**Comment by vlad20012** [comment]
Oh, `RsPath.referenceNameElement` should be nullable now...

File: src/main/kotlin/org/rust/lang/core/psi/ext/RsPath.kt [hunk 1]
```diff
@@ -91,7 +91,7 @@ abstract class RsPathImplMixin : RsStubbedElementImpl<RsPathStub>,
 
     override val referenceNameElement: PsiElement
         get() = checkNotNull(identifier ?: self ?: `super` ?: cself ?: crate) {
-            "Path must contain identifier: $this ${this.text} at ${this.containingFile.virtualFile.path}"
+            "Path must contain identifier: $this ${this.text} at ${this.containingFile.virtualFile?.path}"
```
Comment (ortem) [inline src/main/kotlin/org/rust/lang/core/psi/ext/RsPath.kt]: Now `IllegalStateException` is thrown for every incomplete path (e.g. `Foo::`) in the editor.

Comment (vlad20012) [reply src/main/kotlin/org/rust/lang/core/psi/ext/RsPath.kt]: :+1: 
This is the reason why it's WIP

**Comment by vlad20012** [comment]
I'll try to fix it later

**Pull request closed without merging**